interpreter/llvm/src/tools/clang/lib/Sema/CodeCompleteConsumer.cpp

```diff
@@ -428,15 +429,39 @@ CodeCompleteConsumer::OverloadCandidate::getFunctionType() const {
 
 CodeCompleteConsumer::~CodeCompleteConsumer() { }
 
+bool PrintingCodeCompleteConsumer::isResultFilteredOut(StringRef Filter,
+                                                CodeCompletionResult Result) {
+  switch (Result.Kind) {
+  case CodeCompletionResult::RK_Declaration: {
+    return !(Result.Declaration->getIdentifier() &&
+            (*Result.Declaration).getIdentifier()->getName().startswith(Filter));
+  }
+  case CodeCompletionResult::RK_Keyword: {
+    return !((StringRef(Result.Keyword)).startswith(Filter));
+  }
+  case CodeCompletionResult::RK_Macro: {
+    return !(Result.Macro->getName().startswith(Filter));
+  }
+  case CodeCompletionResult::RK_Pattern: {
+    return !(StringRef((Result.Pattern->getAsString())).startswith(Filter));
+  }
+  default: llvm_unreachable("Unknown code completion result Kind.");
+  }
+}
+
 void 
 PrintingCodeCompleteConsumer::ProcessCodeCompleteResults(Sema &SemaRef,
```

Comment:
Do we also need to re-implement `ProcessOverloadCandidates()` or is this handled as part of `ProcessCodeCompleteResults()` by default?